src/AWS.Deploy.Orchestration/CustomRecipeLocator.cs

```diff
@@ -88,26 +88,21 @@ private async Task<List<string>> LocateRecipePathsFromManifestFile(string target
         }
 
         /// <summary>
-        /// Fetches custom recipe paths from other locations that are monitored by the same source control root
-        /// as the target application that needs to be deployed.
-        /// If the target application is not under source control then it scans the sub-directories of the solution folder for custom recipes.
+        /// Fetches custom recipe paths from other locations that are monitored by the same source control root as the target application that needs to be deployed.
+        /// If the target application is not under source control, then it scans the sub-directories of the solution folder for custom recipes.
+        /// If source control root directory is equal to the file system root, then it scans the sub-directories of the solution folder for custom recipes.
         /// </summary>
         /// <param name="targetApplicationFullPath">The absolute path to the target application csproj or fsproj file</param>
         /// <param name="solutionDirectoryPath">The absolute path of the directory which contains the solution file for the target application</param>
         /// <returns>A list of recipe definition paths.</returns>
         private List<string> LocateAlternateRecipePaths(string targetApplicationFullPath, string solutionDirectoryPath )
         {
             var targetApplicationDirectoryPath = _directoryManager.GetDirectoryInfo(targetApplicationFullPath).Parent.FullName;
-            string? rootDirectoryPath;
+            var fileSystemRootPath = _directoryManager.GetDirectoryInfo(targetApplicationDirectoryPath).Root.FullName;
+            var rootDirectoryPath = GetSourceControlRootDirectory(targetApplicationDirectoryPath);
 
-            if (IsDirectoryUnderSourceControl(targetApplicationDirectoryPath))
-            {
-                rootDirectoryPath = GetSourceControlRootDirectory(targetApplicationDirectoryPath);
-            }
-            else
-            {
+            if (string.IsNullOrEmpty(rootDirectoryPath) || string.Equals(rootDirectoryPath, fileSystemRootPath))
                 rootDirectoryPath = solutionDirectoryPath;
-            }
 
             return GetRecipePathsFromRootDirectory(rootDirectoryPath);
```

Comment:
When `GetRecipePathsFromRootDirectory` calls `_directoryManager.GetFiles`. If that `GetFiles` method fails can you report a logging message and then return an empty collection of paths. This would have allowed Bo during his testing to keep working even though he was having permissioning issues.

Reply:
Fixed